keps/sig-scheduling/5004-dra-extended-resource/README.md

```diff
@@ -0,0 +1,1061 @@
+
+No.
+
+###### Will enabling / using this feature result in increasing size or count of the existing API objects?
+
+Yes. With the extensions proposed in this KEP, individual
+`DeviceClass`  and `Pod` have additional fields, thus increasing
+their overall signature. In addition,  there is the special resource claim for
+extended resource by DRA, there is at most one such claim per pod.
+
+###### Will enabling / using this feature result in increasing time taken by any operations covered by existing SLIs/SLOs?
+
+Yes. The time to allocate a device to a pod with extended resource request will be affected.
+
+###### Will enabling / using this feature result in non-negligible increase of resource usage (CPU, RAM, disk, IO, ...) in any components?
+
+No.
+
+###### Can enabling / using this feature result in resource exhaustion of some node resources (PIDs, sockets, inodes, etc.)?
+
+No.
+
+### Troubleshooting
+
+<!--
+This section must be completed when targeting beta to a release.
+
+For GA, this section is required: approvers should be able to confirm the
+previous answers based on experience in the field.
+
+The Troubleshooting section currently serves the `Playbook` role. We may consider
+splitting it into a dedicated `Playbook` document (potentially with some monitoring
+details). For now, we leave it here.
+-->
+
+###### How does this feature react if the API server and/or etcd is unavailable?
+
+Will be considered for beta.
+
+###### What are other known failure modes?
+
+<!--
+For each of them, fill in the following information by copying the below template:
+  - [Failure mode brief description]
+    - Detection: How can it be detected via metrics? Stated another way:
+      how can an operator troubleshoot without logging into a master or worker node?
+    - Mitigations: What can be done to stop the bleeding, especially for already
+      running user workloads?
+    - Diagnostics: What are the useful log messages and their required logging
+      levels that could help debug the issue?
+      Not required until feature graduated to beta.
+    - Testing: Are there any tests for failure mode? If not, describe why.
+-->
+  - [Pod pending due to extended resource backed by DRA requests no less than 128 devices]
+    - Detection: inspect pod status 'Pending'
+    - Mitigations: reduce the number of devices requested in one extended resource backed by DRA requests
+    - Diagnostics: scheduler logs at level 5 show the reason for the scheduling failure.
+    - Testing: Will be considered for beta.
+
+###### What steps should be taken if SLOs are not being met to determine the problem?
+
+Will be considered for beta.
+
+## Implementation History
+
+- Kubernetes 1.34: KEP accepted.
+
+## Drawbacks
+
+It adds complexity to the scheduler.
+
+## Alternatives
+
+Many different approaches were considered.
+
+Specifically, the following two alternative proposals were considered:
```

Comment:
I don't fully understand a strong enough reason why this has to be implemented in kube-scheduler, i.e., during the scheduling flow. 

Have we considered this: gate the pods with the extended resource request at kube-apiserver, create a separate controller making necessary API calls, and un-gate pods. 
(Basically same as what I'm suggesting at the implicit tolerations discussion: https://github.com/kubernetes/enhancements/issues/5282#issuecomment-2901506804.)